Compilation of opam-installer failed

I am trying to build opam HEAD from source and encountering a compilation issue with opam-installer. I already have opam 2.0.5 (obtained via ubuntu apt install).

```
$ opam config report
# opam config report
# opam-version      2.0.5 
# self-upgrade      no
# system            arch=x86_64 os=linux os-distribution=ubuntu os-version=20.04
# solver            builtin-mccs+glpk
# install-criteria  -removed,-count[version-lag,request],-count[version-lag,changed],-changed
# upgrade-criteria  -removed,-count[version-lag,solution],-new
# jobs              2
# repositories      1 (http) (default repo at 88a179f5)
# pinned            8 (git)
# current-switch    /home/abcxyz/opam

$ git clone https://github.com/ocaml/opam.git

$ cd opam

$ git rev-parse HEAD
3def1a6d1ff102aaa608f5e9170783b96f55069d

$ opam switch create . ocaml-system.4.08.1

$ opam install . --deps-only

////// ETC ETC ///// 
<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
-> installed cmdliner.1.0.4
-> installed conf-m4.1
-> installed conf-perl.1
-> installed ocamlbuild.0.14.0
-> installed ocamlfind.1.8.1
-> installed base-bytes.base
-> installed ocamlgraph.1.8.8
-> installed opam-file-format.2.0.0
-> installed seq.base
-> installed dune.2.5.1
-> installed cppo.1.6.6
-> installed re.1.9.0
-> installed extlib.1.7.7
-> installed cudf.0.9-1
-> installed mccs.1.1+11
-> installed opam-core.2.1.0~alpha
-> installed dose3.5.0.1
-> installed opam-format.2.1.0~alpha
[ERROR] The compilation of opam-installer failed at "/home/abcxyz/.opam/opam-init/hooks/sandbox.sh build make opam-installer.install".
-> installed opam-repository.2.1.0~alpha
-> installed opam-solver.2.1.0~alpha
-> installed opam-state.2.1.0~alpha
-> installed opam-client.2.1.0~alpha
-> installed opam-devel.2.1.0~alpha

#=== ERROR while compiling opam-installer.2.1.0~alpha =========================#
# context     2.0.5 | linux/x86_64 | ocaml-system.4.08.1 | pinned(git+file:///home/abcxyz/opam#master#3def1a6d)
# path        ~/opam/_opam/.opam-switch/build/opam-installer.2.1.0~alpha
# command     ~/.opam/opam-init/hooks/sandbox.sh build make opam-installer.install
# exit-code   2
# env-file    ~/.opam/log/opam-installer-306825-783922.env
# output-file ~/.opam/log/opam-installer-306825-783922.out
### output ###
# dune build --profile=release  --promote-install-files opam-installer.install
# Error: Error: mkdir: _build: Read-only file system
# make: *** [Makefile:113: opam-installer.install] Error 1
<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+- The following actions failed
| - build opam-installer 2.1.0~alpha
+- 
+- The following changes have been performed
| - install base-bytes       base
| - install cmdliner         1.0.4
| - install conf-m4          1
| - install conf-perl        1
| - install cppo             1.6.6
| - install cudf             0.9-1
| - install dose3            5.0.1
| - install dune             2.5.1
| - install extlib           1.7.7
| - install mccs             1.1+11
| - install ocamlbuild       0.14.0
| - install ocamlfind        1.8.1
| - install ocamlgraph       1.8.8
| - install opam-client      2.1.0~alpha
| - install opam-core        2.1.0~alpha
| - install opam-devel       2.1.0~alpha
| - install opam-file-format 2.0.0
| - install opam-format      2.1.0~alpha
| - install opam-repository  2.1.0~alpha
| - install opam-solver      2.1.0~alpha
| - install opam-state       2.1.0~alpha
| - install re               1.9.0
| - install seq              base
+- 

<><> opam-devel.2.1.0~alpha installed successfully ><><><><><><><><><><><><><><>
=> The development version of opam has been successfully compiled into /home/abcxyz/opam/_opam/lib/opam-devel. You should not run it from there, please
   install the binaries to your PATH, e.g. with
       sudo cp /home/abcxyz/opam/_opam/lib/opam-devel/opam /usr/local/bin

   If you just want to give it a try without altering your current installation, you could use instead:
       alias opam2="OPAMROOT=~/.opam2 /home/abcxyz/opam/_opam/lib/opam-devel/opam"
# Run eval $(opam env) to update the current shell environment
```

--- a/Makefile
+++ b/Makefile
@@ -103,9 +103,6 @@ opam-devel.install: $(DUNE_DEP)
 	$(DUNE) build $(DUNE_ARGS) -p opam opam.install
 	sed -e "s/bin:/libexec:/" opam.install > $@
 
-opam-%.install: $(DUNE_DEP)
-	$(DUNE) build $(DUNE_ARGS) -p opam-$* $@
-
 .PHONY: build-opam-installer
 build-opam-installer: $(DUNE_DEP) 
 	$(DUNE) build --profile=$(DUNE_PROFILE) $(DUNE_ARGS)$(DUNE_PROMOTE_ARG) opam-installer.install

--- a/master_changes.md
+++ b/master_changes.md
@@ -4,6 +4,10 @@ note.
 ## Init
   * Remove m4 from the list of recommended tools [#4184 @kit-ty-kate]
 
+## Build
+  * Opam file build using dune, removal of opam-%.install makefile target [#4178 @rjbou - fix #4173]
+  * Use version var in opam file instead of equal current version number in opamlib dependencies [#4178 @rjbou]
+
 ## Switch
   * Fix Not_found with `opam switch create . --deps` [#4151 @AltGr]
 

--- a/opam-client.opam
+++ b/opam-client.opam
@@ -18,7 +18,7 @@ bug-reports: "https://github.com/ocaml/opam/issues"
 dev-repo: "https://github.com/ocaml/opam.git"
 build: [
   ["./configure" "--disable-checks" "--prefix" prefix]
-  [make "%{name}%.install"]
+  ["dune" "build" "-p" name "-j" jobs]
 ]
 depends: [
   "opam-state" {= "2.1.0~alpha"}

--- a/opam-client.opam
+++ b/opam-client.opam
@@ -21,10 +21,10 @@ build: [
   ["dune" "build" "-p" name "-j" jobs]
 ]
 depends: [
-  "opam-state" {= "2.1.0~alpha"}
-  "opam-solver" {= "2.1.0~alpha"}
+  "opam-state" {= version}
+  "opam-solver" {= version}
   "extlib"
-  "opam-repository" {= "2.1.0~alpha"}
+  "opam-repository" {= version}
   "re" {>= "1.9.0"}
   "cmdliner" {>= "0.9.8"}
   "dune" {build & >= "1.2.1"}

--- a/opam-core.opam
+++ b/opam-core.opam
@@ -18,7 +18,7 @@ bug-reports: "https://github.com/ocaml/opam/issues"
 dev-repo: "https://github.com/ocaml/opam.git"
 build: [
   ["./configure" "--disable-checks" "--prefix" prefix]
-  [make "%{name}%.install"]
+  ["dune" "build" "-p" name "-j" jobs]
 ]
 depends: [
   "base-unix"

--- a/opam-devel.opam
+++ b/opam-devel.opam
@@ -18,7 +18,7 @@ bug-reports: "https://github.com/ocaml/opam/issues"
 dev-repo: "https://github.com/ocaml/opam.git"
 build: [
   ["./configure" "--disable-checks" "--prefix" prefix]
-  [make "%{name}%.install"]
+  ["dune" "build" "-p" name "-j" jobs]
 ]
 build-test: [make "tests"]
 depends: [

--- a/opam-devel.opam
+++ b/opam-devel.opam
@@ -22,7 +22,7 @@ build: [
 ]
 build-test: [make "tests"]
 depends: [
-  "opam-client" {= "2.1.0~alpha"}
+  "opam-client" {= version}
   "cmdliner" {>= "0.9.8"}
   "dune" {build & >= "1.2.1"}
 ]

--- a/opam-format.opam
+++ b/opam-format.opam
@@ -21,7 +21,7 @@ build: [
   ["dune" "build" "-p" name "-j" jobs]
 ]
 depends: [
-  "opam-core" {= "2.1.0~alpha"}
+  "opam-core" {= version}
   "opam-file-format" {>= "2.0.0~rc2"}
   "dune" {build & >= "1.2.1"}
 ]

--- a/opam-format.opam
+++ b/opam-format.opam
@@ -18,7 +18,7 @@ bug-reports: "https://github.com/ocaml/opam/issues"
 dev-repo: "https://github.com/ocaml/opam.git"
 build: [
   ["./configure" "--disable-checks" "--prefix" prefix]
-  [make "%{name}%.install"]
+  ["dune" "build" "-p" name "-j" jobs]
 ]
 depends: [
   "opam-core" {= "2.1.0~alpha"}

--- a/opam-installer.opam
+++ b/opam-installer.opam
@@ -18,7 +18,7 @@ bug-reports: "https://github.com/ocaml/opam/issues"
 dev-repo: "https://github.com/ocaml/opam.git"
 build: [
   ["./configure" "--disable-checks" "--prefix" prefix]
-  [make "%{name}%.install"]
+  ["dune" "build" "-p" name "-j" jobs]
 ]
 depends: [
   "opam-format" {= "2.1.0~alpha"}

--- a/opam-installer.opam
+++ b/opam-installer.opam
@@ -21,7 +21,7 @@ build: [
   ["dune" "build" "-p" name "-j" jobs]
 ]
 depends: [
-  "opam-format" {= "2.1.0~alpha"}
+  "opam-format" {= version}
   "cmdliner" {>= "0.9.8"}
   "dune" {build & >= "1.2.1"}
 ]

--- a/opam-repository.opam
+++ b/opam-repository.opam
@@ -18,7 +18,7 @@ bug-reports: "https://github.com/ocaml/opam/issues"
 dev-repo: "https://github.com/ocaml/opam.git"
 build: [
   ["./configure" "--disable-checks" "--prefix" prefix]
-  [make "%{name}%.install"]
+  ["dune" "build" "-p" name "-j" jobs]
 ]
 depends: [
   "opam-format" {= "2.1.0~alpha"}

--- a/opam-repository.opam
+++ b/opam-repository.opam
@@ -21,7 +21,7 @@ build: [
   ["dune" "build" "-p" name "-j" jobs]
 ]
 depends: [
-  "opam-format" {= "2.1.0~alpha"}
+  "opam-format" {= version}
   "dune" {build & >= "1.2.1"}
 ]
 available: ocaml-version >= "4.02.3"

--- a/opam-solver.opam
+++ b/opam-solver.opam
@@ -18,7 +18,7 @@ bug-reports: "https://github.com/ocaml/opam/issues"
 dev-repo: "https://github.com/ocaml/opam.git"
 build: [
   ["./configure" "--disable-checks" "--prefix" prefix]
-  [make "%{name}%.install"]
+  ["dune" "build" "-p" name "-j" jobs]
 ]
 depends: [
   "opam-format" {= "2.1.0~alpha"}

--- a/opam-solver.opam
+++ b/opam-solver.opam
@@ -21,7 +21,7 @@ build: [
   ["dune" "build" "-p" name "-j" jobs]
 ]
 depends: [
-  "opam-format" {= "2.1.0~alpha"}
+  "opam-format" {= version}
   "mccs" {>= "1.1+9"}
   "dose3" {>= "5"}
   "cudf" {>= "0.7"}

--- a/opam-state.opam
+++ b/opam-state.opam
@@ -18,7 +18,7 @@ bug-reports: "https://github.com/ocaml/opam/issues"
 dev-repo: "https://github.com/ocaml/opam.git"
 build: [
   ["./configure" "--disable-checks" "--prefix" prefix]
-  [make "%{name}%.install"]
+  ["dune" "build" "-p" name "-j" jobs]
 ]
 depends: [
   "opam-repository" {= "2.1.0~alpha"}

--- a/opam-state.opam
+++ b/opam-state.opam
@@ -21,7 +21,7 @@ build: [
   ["dune" "build" "-p" name "-j" jobs]
 ]
 depends: [
-  "opam-repository" {= "2.1.0~alpha"}
+  "opam-repository" {= version}
   "dune" {build & >= "1.2.1"}
 ]
 available: ocaml-version >= "4.02.3"

